server/src/main/java/org/opensearch/index/engine/InternalEngine.java

```diff
@@ -2587,6 +2655,11 @@ protected void commitIndexWriter(final IndexWriter writer, final String translog
                 return commitData.entrySet().iterator();
             });
             shouldPeriodicallyFlushAfterBigMerge.set(false);
+            if (isContextAwareEnabled) {
+                // To sync document during commit. This keeps documents during commit always ahead of checkpoint.
+                refresh("commit", SearcherScope.INTERNAL, true);
```

Comment:
why is this in internal scope? Post commit, are we saying that docs will still not be available for user reads?